src/ert/ensemble_evaluator/_builder/_legacy.py

```diff
@@ -199,46 +194,35 @@ async def _evaluate_inner(  # pylint: disable=too-many-branches
                     )
                 ]
 
-            # Tell queue to pass info to the jobs-file
-            # NOTE: This touches files on disk...
-            sema = threading.BoundedSemaphore(value=CONCURRENT_INTERNALIZATION)
-            self._job_queue.add_dispatch_information_to_jobs_file(
+            self._job_queue.set_ee_info(
+                ee_uri=self._config.dispatch_uri,
                 ens_id=self.id_,
-                dispatch_url=self._config.dispatch_uri,
-                cert=self._config.cert,
-                token=self._config.token,
-            )
-            # Finally, run the queue-loop until it finishes or raises
-            await self._job_queue.execute_queue_via_websockets(
-                self._config.dispatch_uri,
-                self.id_,
-                sema,
-                queue_evaluators,  # type: ignore
                 ee_cert=self._config.cert,
                 ee_token=self._config.token,
             )
 
-        except asyncio.CancelledError:
-            logger.debug("ensemble was cancelled")
-            result = event_creator(identifiers.EVTYPE_ENSEMBLE_CANCELLED, None)
+            # Tell queue to pass info to the jobs-file
+            # NOTE: This touches files on disk...
+            self._job_queue.add_dispatch_information_to_jobs_file()
+
+            sema = threading.BoundedSemaphore(value=CONCURRENT_INTERNALIZATION)
+            result: str = await self._job_queue.execute(
+                sema,
+                queue_evaluators,
+            )
 
         except Exception:
             logger.exception(
                 "unexpected exception in ensemble",
                 exc_info=True,
             )
-            result = event_creator(identifiers.EVTYPE_ENSEMBLE_FAILED, None)
+            result = identifiers.EVTYPE_ENSEMBLE_FAILED
 
-        else:
-            logger.debug("ensemble finished normally")
-
-        finally:
-            await timeout_queue.put(None)  # signal to exit timer
-            await send_timeout_future
+        await timeout_queue.put(None)  # signal to exit timer
```

Comment:
I see that you've removed the finally section. Isn't point of finally that it gets always executed even when there was an exception?

Reply:
Yes, but since we capture absolutely every exception, what comes after the `try:` block will get executed nonetheless.

Reply:
Right, I've missed the `except Exception:`. Then it's fine I guess.